src/components/KTruncate/KTruncate.vue

```diff
@@ -0,0 +1,280 @@
+            <KIcon
+              color="var(--KTruncateCollapseIconColor, var(--blue-500, color(blue-500)))"
+              icon="chevronUp"
+              size="10"
+              title="Show less"
+            />
+          </KButton>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="!isTextContent && showToggle"
+      class="d-flex h-100 align-items-end"
+    >
+      <div
+        v-if="!expanded"
+        data-testid="expand-trigger-wrapper"
+      >
+        <slot
+          :excess-elements-count="excessElementsCount"
+          :expand="handleToggleClick"
+          name="expand-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="expand-trigger"
+            @click="handleToggleClick"
+          >
+            {{ excessElementsCount }}
+          </KButton>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="isTextContent && (showToggle || expanded)"
+      ref="textToggleControls"
+      class="place-self-end mt-2"
+    >
+      <div
+        v-if="!expanded"
+        data-testid="expand-trigger-wrapper"
+      >
+        <slot
+          :expand="handleToggleClick"
+          name="expand-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="expand-trigger"
+            @click="handleToggleClick"
+          >
+            Show more
+          </KButton>
+        </slot>
+      </div>
+      <div v-if="expanded">
+        <slot
+          :collapse="handleToggleClick"
+          name="collapse-trigger"
+        >
+          <KButton
+            appearance="btn-link"
+            class="collapse-trigger"
+            @click="handleToggleClick"
+          >
+            Show less
+          </KButton>
+        </slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, onBeforeUnmount, ref, nextTick } from 'vue'
+
+const props = defineProps({
+  rows: {
+    type: Number,
+    default: 1,
+    validator: (value: Number): boolean => value > 0,
+  },
+  isTextContent: {
+    type: Boolean,
+    default: false,
+  },
+  isExpanded: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const expanded = ref<boolean>(props.isExpanded)
+
+const showToggle = ref<boolean>(false)
+
+const resizeObserver = ref()
+
+const kTruncateContainer = ref<HTMLDivElement>()
+const kTruncateWrapper = ref<HTMLDivElement>()
+const textToggleControls = ref<HTMLDivElement>()
+const wrapperHeight = ref<string>('0px')
+
+const excessElementsCount = ref<number>(0)
+
+/**
+ * Sets wrapper height.
+ * When rows prop === 1: sets wrapper height equal to height of the tallest child.
+ * When rows prop > 1: wrapper hight will be rows count multiplied by height of the tallest child plus gap (10px).
+ * For example if rows is 2 and all elements are equal height if 22px, wrapper height will be set to 54px (2 * 22 + 10).
+ */
+const setWrapperHeight = (): void => {
+  if (props.isTextContent) {
+    return
+  }
```

Comment:
question: since the prop values can change, instead of just returning here, should you reset the `wrapperHeight.value` back to the default `0px`?

Reply:
I'm not sure I understand what you mean. The return here is needed rather for optimization purposes - to avoid doing those calculations because they are not needed when `props.isTextContent === true`, the `wrapperHeight.value` is not taken into account in such case

Reply:
Right but what I'm asking is if a scenario like this is covered:

1. The user initializes the component with `isTextContent="false"` and so then the `wrapperHeight.value` is set to something that is not `0px`
2. In the host app, the slot content is changed dynamically and the `isTextContent` prop value is now switched to `true`

This means that the logic in this block would be ignored, but the `wrapperHeight.value` is still set to a value other than `0px`. Does this matter?